### PR TITLE
#191 Draft API for sharing docs

### PR DESCRIPTION
**Pull request by longtimeago**
#191 Draft API for sharing docs

Basic idea is to use Invitations for creating links to be sent by email and  accepting invitations.

@yegor256 I understand that:
- there's no tests
- there's no real value
- not a top-down approach

But I'm really need to understand am I on the right track?
If yes, than what do you think will be useful to add in the PR apart from several puzzles.
I have 10-15 mins left

### CODE REVIEW

**Comment by karato** [comment]
@longtimeago Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by karato** [comment]
@darkled review it

File: src/main/java/com/nerodesk/om/mock/MkInvitations.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * @author Paul Polishchuk (ppol@ua.fm)
+ * @version $Id$
+ * @since 0.4
+ */
+public final class MkInvitations implements Invitations {
+
+    @Override
+    public void send(final User owner, final String doc, final String email) {
+        throw new UnsupportedOperationException("MkInvitations#send()");
```
Comment (asaen) [inline src/main/java/com/nerodesk/om/mock/MkInvitations.java:46]: @longtimeago do we need a puzzle here?

File: src/main/java/com/nerodesk/om/aws/AwsInvitations.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Override
+    // @checkstyle ParameterNumberCheck (5 lines)
+    public void accept(
+        final User friend,
+        final String doc,
+        final String owner,
+        final String code
+    ) {
+        throw new UnsupportedOperationException("AwsInvitations#accept()");
```
Comment (asaen) [inline src/main/java/com/nerodesk/om/aws/AwsInvitations.java:57]: @longtimeago do we need a puzzle here?

File: src/main/java/com/nerodesk/om/aws/AwsInvitations.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * @author Paul Polishchuk (ppol@ua.fm)
+ * @version $Id$
+ * @since 0.4
+ */
+public final class AwsInvitations implements Invitations {
+
+    @Override
+    public void send(final User owner, final String doc, final String email) {
+        throw new UnsupportedOperationException("AwsInvitations#send()");
```
Comment (asaen) [inline src/main/java/com/nerodesk/om/aws/AwsInvitations.java:46]: @longtimeago do we need a puzzle here?

**Comment by longtimeago** [comment]
@darkled that's why I asked for Yegor's review ... not actually the review but advice do define if I'm going right direction.
Yes, I understand that we need puzzles there. But it's only if the whole approach is right.

PR is not finished, it's a draft

**Comment by asaen** [comment]
@longtimeago ok, got it.

**Comment by longtimeago** [comment]
@yegor256 please, see PR description

**Comment by longtimeago** [comment]
@darkled sorry for this misleading situation

**Comment by asaen** [comment]
@rultor merge

**Comment by k0r** [comment]
@yegor256 postpone merging this.

**Comment by asaen** [comment]
@yegor256 will this postponed PR stay as it is or should we close it somehow?

**Comment by yegor256** [comment]
@darkled let's keep it open for some time. I'll update soon

**Comment by asaen** [comment]
@yegor256 any update here? The PR seems to be opened for too long time...

**Comment by yegor256** [comment]
@longtimeago I think we should close it, since this feature is not required any more

**Comment by karato** [comment]
@darkled thanks, paid, **19 mins** to your account, payment ID is `63984270`, task took 2878 hours and 58 mins to complete

extra minutes for review comments (c=4)

+19 to your rating, your total score is [+285](http://www.netbout.com/b/35091?open=rating)

**Pull request closed without merging**